libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/NotificationType.swift

```diff
@@ -55,21 +55,48 @@ public class Action: NSObject, Codable {
     public let name: String
     public let identifier: String
     public let label: String
-    public let type: String
+    public let type: ActionType
     
     enum CodingKeys: String, CodingKey {
         case name, label, type
         case identifier = "actionKey"
     }
     
-    public init(name: String, identifier: String, label: String, type: String) {
+    public init(name: String, identifier: String, label: String, type: ActionType) {
         self.name = name
         self.identifier = identifier
         self.label = label
         self.type = type
     }
 }
 
+@objc(SFSDKActionType)
```

Comment:
Added additional processing support to check the Action Type.  For now, we default to a Foreground Action, but if it's an Actionable Notification type, we'll make sure that it's not foregrounded.  Here we could add addtionable Action Types for this if we need to in the future by adding addtional cases in to ActionType  enum.